symengine/assumptions.cpp

```diff
@@ -27,10 +28,86 @@ Assumptions::Assumptions(const set_basic &statements)
                     integer_symbols_.insert(expr);
                 }
             }
+        } else if (is_a<LessThan>(*s)) {
+            const LessThan &less_than = down_cast<const LessThan &>(*s);
+            const auto &arg1 = less_than.get_arg1();
+            const auto &arg2 = less_than.get_arg2();
+            if (is_a<Symbol>(*arg2) and is_a_Number(*arg1)) {
+                real_symbols_.insert(arg2);
+                if (down_cast<const Number &>(*arg1).is_positive()) {
+                    set_map(nonnegative_, arg2, true);
+                    set_map(positive_, arg2, true);
+                    set_map(negative_, arg2, false);
+                    set_map(nonpositive_, arg2, false);
```

Comment:
lgtm!

One question - here `nonpositive_` seems redundant, since whenever it is set, `positive_` is also set to the opposite value - similarly for `nonnegative_` and `negative`. Would it be simpler to only have `positive_`, `negative_`, and add an extra line of logic in `is_nonpositive()` and `is_nonnegative()`? Or would that potentially make things more complicated as more assumptions are added later?

Reply:
Since `Basic` can be more "things" than numbers a symbol can be both not positive and not nonpositive at the same time. That's why I implemented it this way. If we had a structure with no mixing with sets, booleans, logic etc I think it could be implemented differently.

Reply:
I just realized that a complex number also is both not positive and not nonpositive at the same time...

Reply:
Ok makes sense, thanks!